modules/ROOT/pages/for-each-scope-concept.adoc

```diff
@@ -22,6 +22,37 @@ Note that if the input contains information outside the collection you tell it t
 
 You can also split an array in batches. Potentially, batches produce quicker processing. Each batch is treated as a separate Mule message. For example, if a collection has 200 elements and you set Batch Size (`batchSize`) to `50`, the For Each scope iteratively processes 4 batches of 50 elements, each as a separate Mule message.
 
+== Variable Propagation
+
+Every execution of the ForEach scope will start with the same variables and values as before the execution of the block. New variables or modifications of already existing ones while processing one element will not be visible while processing another element. All of those variable changes will not be avaiable outside the ForEach scope. In other words, the set of variables (and their values) after the execution of the ForEach Scope will remain the same as before the execution of it.
```

Comment:
`the execution of it` -> `its execution`?

Reply:
```suggestion
Every execution of the ForEach scope will start with the same variables and values as before the execution of the block. New variables or modifications of already existing ones while processing one element will not be visible while processing another element. All of those variable changes will not be avaiable outside the ForEach scope. In other words, the set of variables (and their values) after the execution of the ForEach Scope will remain the same as before its execution.
```